domains/client/domain-executor/src/system_bundle_processor.rs

```diff
@@ -114,7 +114,13 @@ where
         let parent_hash = self.client.info().best_hash;
         let parent_number = self.client.info().best_number;
 
-        let extrinsics = self.bundles_to_extrinsics(parent_hash, bundles, shuffling_seed)?;
+        let mut extrinsics = self.bundles_to_extrinsics(parent_hash, bundles, shuffling_seed)?;
+        if let Some(primary_hash_extrinsic) =
+            self.construct_submit_primary_hash_extrinsic(parent_hash, primary_number, primary_hash)?
+        {
+            // Push the `submit_primary_hash` extrinsic to the end so that it will take effect for the next block
+            extrinsics.push(primary_hash_extrinsic);
+        }
```

Comment:
This PR maintains the `BlockHash` for core domains by introducing a new `submit_primary_hash` extrinsic and submitting this extrinsic while the system domain process bundles (similar to `submit_core_bundle`). 

The `(primary_number, primary_hash)` used is actually the same as the current implement I think, but this PR only maintains one `BlockHash` state for all the core domains instead of one for each core domain. /cc @liuchengxu

Reply:
Using a shared `BlockHash` for all domains is incorrect, let's say a core domain runs into some issue and stalls, but the other domains move forward regularly, once the `BlockHash` is pruned due to the other workings domains reaching the pruning depth, the state of the stalled domain will be corrupted, thus `BlockHash` for each domain is necessary.